src/components/App/App.js

```diff
@@ -50,29 +50,26 @@ const App = () => {
     <div className="App">
       <NavBar />
       <Switch>
+
         <Route exact path="/">
           {error.length > 0 ? 
-          <Error message={error}/> 
+            <Error message={error}/> 
           : 
-          <TileBucket allMovies={allMovieTiles} getData={getData} />}
+            <TileBucket allMovies={allMovieTiles} getData={getData} />}
         </Route>
-        <Route path="/movies/:anything" render={({match})=> {
-          if (selectedMovie) {
-            return <MovieDeets selectedMovie={selectedMovie} getData={getData} match={match}/>
+
+        <Route path="/movies/:id" render={({match})=> {
+          if (selectedMovie === '') {
+            getData(match.params.id)
```

Comment:
I think it's much less complicated to remove the useEffect from MovieDeets and only allow App to delegate the data